### ISSUE TITLE
Callback from MobileMessaging.ResultListener<Installation> class is not receiving for Samsung M31

### ISSUE THREAD

**Issue by pratikkukadia** [infobip/mobile-messaging-sdk-android#28]
Hello,

When we are going to activate infobip push notification that time for device which mention below, in that infobip is not giving callback in onResult().

**Current Infobip SDK version: 4.3.1
Device: Samsung M31
Android O.S version: Android 11**

* Code snippet:
```
class InfoBipSDKController : MobileMessaging.ResultListener<Installation>() {
    init {
        activateInfobipNotification()
    }
    override fun onResult(result: Result<Installation, MobileMessagingError>?) {
        if (result?.isSuccess == true) {
            // Success
        } else {
            // Failure
        }
    }
    fun activateInfobipNotification() {
        // Infobip Registration code.
    }
}
```

**Comment by riskpp** [comment]
Hi @pratikkukadia , could you please update Mobile Messaging SDK to the latest version - [5.2.5](https://github.com/infobip/mobile-messaging-sdk-android/releases/tag/5.2.5)

BR, Olga

**Comment by pratikkukadia** [comment]
Hi @riskpp ,

Updated the Mobile Messaging SDK to the latest version - 5.2.5

But still the issue persist.

**Comment by riskpp** [comment]
Hi @pratikkukadia,

Could you please provide more details about your use-case? On the provided code-snippet I don't see that this listener is used somehow. Usually `MobileMessaging.ResultListener` is provided to some method as in [fetching installation example](https://github.com/infobip/mobile-messaging-sdk-android/wiki/Users-and-installations#installation).

If you want to receive the [REGISTRATION_CREATED event](https://github.com/infobip/mobile-messaging-sdk-android/wiki/Library-events#registration_created) or the [INSTALLATION_UPDATED event](https://github.com/infobip/mobile-messaging-sdk-android/wiki/Library-events#installation_updated),
you could create `BroadcastReceiver` and register it as in examples from [Library events guide](https://github.com/infobip/mobile-messaging-sdk-android/wiki/Library-events)

BR, Olga

**Comment by pratikkukadia** [comment]
Hi @riskpp 

We first initialized the Infobip SDK using following code:

 ```
fun initializeInfobipSDK() {
        val mobileMessaging = MobileMessaging.getInstance(context)

        MobileMessaging.Builder(context)
            .withHttpSettings(true)
            .withDisplayNotification(
                NotificationSettings.Builder(context)
                    .withMultipleNotifications()
                    .withDefaultIcon(R.drawable.ic_notificaiton)
                    .withColor(ContextCompat.getColor(context, R.color.colorAccent))
                    .build()
            )
            .build(initInfobipListener)

    }


Then we perform saveInstallation using following code:

private fun setInstallationAttribute() {
        val mobileMessaging = MobileMessaging.getInstance(context)
        val installation = mobileMessaging.installation
        installation.isPushRegistrationEnabled = true

        val customUserDataValue = CustomAttributeValue("true")
        installation.setCustomAttribute(serviceNotificationSegment, customUserDataValue)
        installation.setCustomAttribute(transactionSegment, customUserDataValue)
        mobileMessaging.saveInstallation(installation, this)
    }    

```

Once above process gets completed then we expect callback on onResult of ResultListener<Installation> listener


override fun onResult(result: Result<Installation, MobileMessagingError>?) {
        // For Device: Samsung M31 (Android O.S version: Android 11) not getting callback
    }  

So in this case process is endless. So until we do not receive result callback our application gets stuck.

**Comment by riskpp** [comment]
Hello @pratikkukadia, could you please provide logs to us?
Is"Samsung M31" it the only device where this code doesn't work?

We've tried to create custom listener and call `saveInstallation` right inside it and haven't reproduce any issues.

Here it's the code for your reference:

MainActivity.kt
```kt
class MainActivity : AppCompatActivity() {
    val customListener = CustomListener()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val initListener = MyInitListener()
        MobileMessaging.Builder(application)
                .withHttpSettings(true)
                .withDisplayNotification(
                        NotificationSettings.Builder(application)
                                .withMultipleNotifications()
                                .build()
                )
                .build(initListener)

        customListener.updateInstallation(application)
    }
}

class MyInitListener: MobileMessaging.InitListener {

    override fun onSuccess() {
        Log.d("TestMM", "Initialization Success")
    }
    override fun onError(e: InternalSdkError?, googleErrorCode: Int?) {
        Log.d("TestMM", "Initialization Error: " + e.toString())
    }

}
```

CustomListener.kt:
```kt
class CustomListener : MobileMessaging.ResultListener<Installation>() {
    override fun onResult(result: Result<Installation, MobileMessagingError>?) {
        Log.d("TestMM", "onResult error: " + result?.error?.message + ", data: " + result?.data?.pushRegistrationId)
    }

    public fun updateInstallation(context: Context) {
        val mobileMessaging = MobileMessaging.getInstance(context)
        val installation = mobileMessaging.installation
        installation.isPushRegistrationEnabled = true

        installation.setCustomAttribute("serviceNotificationSegment", CustomAttributeValue("true"))
        installation.setCustomAttribute("transactionSegment", CustomAttributeValue("true"))
        mobileMessaging.saveInstallation(installation, this)
    }
}
```

**Comment by pratikkukadia** [comment]
Hi @riskpp 

I have checked on another Samsung device S8 Android Version 9 and found the same issue as in Samsung M21. 
Surprisingly in Samsung S8 it was working fine 2 days ago now its not working while in other device its working fine.

Please check the 2 logs for working as well as not working devices

### Working in Nokia 4.2 Android OS Version 10
```
2021-08-31 11:23:33.615 21277-21277/ W/MobileMessaging: Enabled com.google.firebase.iid.FirebaseInstanceIdReceiver for compatibility reasons
2021-08-31 11:23:33.615 21277-21277/ W/MobileMessaging: Enabled org.infobip.mobile.messaging.cloud.firebase.MobileMessagingFirebaseService for compatibility reasons
2021-08-31 11:23:33.677 21277-21277/ V/MobileMessaging: FirebaseRegistrationTokenHandler
    [
      "RECEIVED FCM TOKEN",
      [
        "###########################################################"
      ]
    ]
2021-08-31 11:23:33.694 21277-21277/ W/MobileMessaging: Registration is not available yet
2021-08-31 11:23:33.700 21277-21277/ W/MobileMessaging: Registration is not available yet
2021-08-31 11:23:33.703 21277-21277/ W/MobileMessaging: Registration not available yet, will patch messages later
2021-08-31 11:23:33.703 21277-21563/ W/MobileMessaging: Registration is not available yet
2021-08-31 11:23:33.705 21277-21277/ W/g.mobilebankin: Accessing hidden field Landroid/app/ActivityThread;->mServices:Landroid/util/ArrayMap; (greylist, reflection, allowed)
2021-08-31 11:23:33.706 21277-21563/ V/MobileMessaging: CREATE INSTALLATION >>>
    {
      "appVersion": "18.0",
      "deviceManufacturer": "HMD Global",
      "deviceModel": "Nokia 4.2",
      "deviceName": "Nokia 4.2",
      "deviceSecure": true,
      "deviceTimezoneOffset": "GMT+05:30",
      "geoEnabled": false,
      "language": "en",
      "notificationsEnabled": true,
      "os": "Android",
      "osVersion": "10",
      "pushServiceToken": "###########################################################",
      "pushServiceType": "Firebase",
      "sdkVersion": "4.3.1",
      "map": {
        "appVersion": "18.0",
        "os": "Android",
        "geoEnabled": false,
        "pushServiceToken": "###########################################################",
        "notificationsEnabled": true,
        "language": "en",
        "deviceName": "Nokia 4.2",
        "pushServiceType": "Firebase",
        "osVersion": "10",
        "deviceTimezoneOffset": "GMT+05:30",
        "deviceSecure": true,
        "sdkVersion": "4.3.1",
        "deviceModel": "Nokia 4.2",
        "deviceManufacturer": "HMD Global"
      }
    }
2021-08-31 11:23:33.717 21277-21563/ D/MMHTTP: REQUEST: Request(httpMethod=POST, uri=https://mobile.infobip.com/mobile/1/appinstance, apiKey=################################, credentials=null, headers={applicationcode=[##############], pushregistrationid=[null], foreground=[true], installationid=[################################]}, queryParams={ri=[true]}, body=AppInstance(pushRegId=null, regEnabled=null, notificationsEnabled=true, geoEnabled=false, sdkVersion=4.3.1, appVersion=18.0, os=Android, osVersion=10, deviceManufacturer=HMD Global, deviceModel=Nokia 4.2, deviceSecure=true, language=en, deviceTimezoneOffset=GMT+05:30, applicationUserId=null, deviceName=Nokia 4.2, customAttributes=null, isPrimary=null, pushServiceType=Firebase, pushServiceToken=###########################################################))
```

### Not working in Samsung M31 Android version 11 and Samsung S8 Android version 9

```2021-08-31 11:29:45.751 18106-18106/ W/MobileMessaging: Enabled com.google.firebase.iid.FirebaseInstanceIdReceiver for compatibility reasons
2021-08-31 11:29:45.755 18106-18106/ W/MobileMessaging: Enabled org.infobip.mobile.messaging.cloud.firebase.MobileMessagingFirebaseService for compatibility reasons
2021-08-31 11:29:45.821 18106-18106/ V/MobileMessaging: FirebaseRegistrationTokenHandler
    [
      "RECEIVED FCM TOKEN",
      [
        "###########################################################"
      ]
    ]
2021-08-31 11:29:45.844 18106-19083/ V/MobileMessaging: SYNC MESSAGES >>>
    null
2021-08-31 11:29:45.849 18106-19083/ D/MMHTTP: REQUEST: Request(httpMethod=POST, uri=https://mobile.infobip.com/mobile/5/messages, apiKey=###########################################################, credentials=null, headers={applicationcode=[#########], pushregistrationid=[###############################], foreground=[true], sessionId=[#############################################], installationid=[###############################################]}, queryParams={platformType=[GCM]}, body=null)```

If you need anything else please let us know

Thanks

**Comment by riskpp** [comment]
Hi @pratikkukadia,

Unfortunately logs, you provided, don't fully demonstrate issue for us. 

I'd propose following for you:
1. Provide sample code to us, so we will be able to run it and reproduce the issue.
2. Contact us over support - support@infobip.com, referring to this topic and provide sensitive information over https://s-pass.app.

**Comment by pratikkukadia** [comment]
Hi @riskpp ,

We have followed the steps of installation as mentioned here(https://github.com/infobip/mobile-messaging-sdk-android/wiki/Users-and-installations),
Kindly please try to reproduce this issue on Samsung M31 with android O.S version(11).

![Infobip](https://user-images.githubusercontent.com/12750620/131787425-03291cdc-e620-43bc-a3ad-0f19fad9a5fc.png)

Please provide solution ASAP as we are block with the production issue.

**Comment by riskpp** [comment]
Closing, because issue was resolved in [5.3.2 SDK version](https://github.com/infobip/mobile-messaging-sdk-android/releases/tag/5.3.2)
